### ISSUE TITLE
Necessary brackets for unnest function in BigQuery are removed

### ISSUE THREAD

**Issue by sherif-ghonaim** [sqlfluff/sqlfluff#6142]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I’ve encountered another issue that seems specific to BigQuery involving unnesting within a function. BigQuery requires brackets in this scenario, but SQLFLUFF removes them

### Expected Behaviour

The bracket in this case should not be removed in order not to introduce an error

### Observed Behaviour

I get the attached error when the brackets are removed
<img width="1153" alt="Screenshot 2024-09-04 at 8 45 45 AM" src="https://github.com/user-attachments/assets/820dfb69-724e-42dc-9126-176d1d01ff20">

### How to reproduce

I ran the sqlfluff fix command. Below is the query before and after running the command

Before query:
```
select
    COUNT(
        DISTINCT
        (
            SELECT ANY_VALUE(id)
            FROM UNNEST(tag) t
        )
    )
from
    dataset_name.table_name;
```
After query:
```
SELECT
    COUNT(
        DISTINCT
        SELECT ANY_VALUE(id)
        FROM UNNEST(tag) t

    )
FROM
    dataset_name.table_name;
```

### Dialect

BigQuery

### Version

3.1.1

### Configuration

```cfg
[sqlfluff]

# Supported dialects https://docs.sqlfluff.com/en/stable/dialects.html
# Or run 'sqlfluff dialects'
dialect = bigquery

# One of [raw|jinja|python|placeholder]
templater = jinja
large_file_skip_byte_limit = 0
ignore = templating

# Comma separated list of rules to exclude, or None
# See https://docs.sqlfluff.com/en/stable/configuration.html#enabling-and-disabling-rules
# AM04 (ambiguous.column_count) and ST06 (structure.column_order) are
# two of the more controversial rules included to illustrate usage.
exclude_rules = ambiguous.column_count, structure.column_order, structure.using, capitalisation.identifiers, structure.subquery, structure.join_condition_order

# The standard max_line_length is 80 in line with the convention of
# other tools and several style guides. Many projects however prefer
# something a little longer.
# Set to zero or negative to disable checks.
max_line_length = 180

# CPU processes to use while linting.
# The default is "single threaded" to allow easy debugging, but this
# is often undesirable at scale.
# If positive, just implies number of processes.
# If negative or zero, implies number_of_cpus - specified_number.
# e.g. -1 means use all processors but one. 0 means all cpus.
processes = 0

[sqlfluff:indentation]
# While implicit indents are not enabled by default. Many of the
# SQLFluff maintainers do use them in their projects.
indented_joins = True
indented_using_on = True
allow_implicit_indents = True
indented_on_contents = True
indented_ctes = True
indented_then = True

[sqlfluff:rules:layout.set_operators]
set_operator_on_new_line = ['UNION', 'UNION ALL']

[sqlfluff:rules:convention.statement_brackets]
convention.statement_brackets = True
[sqlfluff:convention.quoted_literals]
force_enable = True
preferred_quoted_literal_style = double_quote

# The default configuration for aliasing rules is "consistent"
# which will auto-detect the setting from the rest of the file. This
# is less desirable in a new project and you may find this (slightly
# more strict) setting more useful.
[sqlfluff:rules:aliasing.table]
aliasing = implicit
[sqlfluff:rules:aliasing.column]
aliasing = explicit
# [sqlfluff:rules:aliasing.length]
# min_alias_length = 10

# The default configuration for capitalisation rules is "consistent"
# which will auto-detect the setting from the rest of the file. This
# is less desirable in a new project and you may find this (slightly
# more strict) setting more useful.
# Typically we find users rely on syntax highlighting rather than
# capitalisation to distinguish between keywords and identifiers.
# Clearly, if your organisation has already settled on uppercase
# formatting for any of these syntax elements then set them to "upper".
# See https://stackoverflow.com/questions/608196/why-should-i-capitalize-my-sql-keywords-is-there-a-good-reason
[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper
[sqlfluff:rules:capitalisation.functions]
extended_capitalisation_policy = upper
[sqlfluff:rules:capitalisation.literals]
capitalisation_policy = upper
[sqlfluff:rules:capitalisation.types]
extended_capitalisation_policy = upper
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by sherif-ghonaim** [comment]
I need to highlight that this issue is not related to distinct, it's the same case without the distinct as well, meaning that the brackets are required even if there is no distinct keyword, not sure if that makes a difference on your solution.

**Comment by keraion** [comment]
@sherif-ghonaim mind throwing a quick example I can test with?

**Comment by alanmcruickshank** [comment]
I'm going to guess that the best similar example would be based on the OP:

```sql
select
    COUNT(
        (
            SELECT ANY_VALUE(id)
            FROM UNNEST(tag) t
        )
    )
from
    dataset_name.table_name;
```

Which I don't think would be affected by ST08, because that rule is explicitly about treating `DISTINCT` as a function.

**Comment by keraion** [comment]
Agreed. I just wanted to make sure OP didn't have another example of a breaking fix without `DISTINCT`.

We can probably move forward with the PR and close this issue. If something else resurfaces we can reopen, and I'll gladly take a look!

**Comment by sherif-ghonaim** [comment]
Hello All,
Sorry just saw your comments now, the adjusted query shared by [alanmcruickshank](https://github.com/alanmcruickshank) is a valid example. Just wondering if the fix would address it or it's handling only in case of distinct keyword is present?
